deltachat-ffi/deltachat.h

```diff
@@ -463,6 +463,64 @@ char*           dc_get_info                  (const dc_context_t* context);
 char*           dc_get_oauth2_url            (dc_context_t* context, const char* addr, const char* redirect_uri);
 
 
+#define DC_CONNECTIVITY_NOT_CONNECTED        1000
+#define DC_CONNECTIVITY_CONNECTING           2000
+#define DC_CONNECTIVITY_WORKING              3000
+#define DC_CONNECTIVITY_INTERRUPTING_IDLE    4000
+#define DC_CONNECTIVITY_CONNECTED            5000
+
+
+/**
+ * Get the current connectivity, i.e. whether the device is connected to the IMAP server.
+ * One of:
+ * - DC_CONNECTIVITY_NOT_CONNECTED (1000-1999): Show e.g. the string "Not connected" or a red dot
+ * - DC_CONNECTIVITY_CONNECTING (2000-2999): Show e.g. the string "Connecting…" or a yellow dot
+ * - DC_CONNECTIVITY_WORKING (3000-3999): Show e.g. the string "Getting new messages" or a spinning wheel
+ * - DC_CONNECTIVITY_INTERRUPTING_IDLE or DC_CONNECTIVITY_CONNECTED (>=4000): Show e.g. the string "Connected" or a green dot
+ *
+ * We don't use exact values but ranges here so that we can split up
+ * states into multiple states in the future.
+ *
+ * Meant as a rough overview that can be shown 
+ * e.g. in the title of the main screen.
+ *
+ * Also, you can use this to find out when the core is completely done with fetching:
+ * - call dc_start_io() (in case IO was not running)
+ * - call dc_maybe_network()
+ * - wait until the connectivity is DC_CONNECTIVITY_CONNECTED (>=5000)
+ * That's also the reason why we have DC_CONNECTIVITY_INTERRUPTING_IDLE:
+ * When you call dc_maybe_network, the connectivity is immediately set to
+ * DC_CONNECTIVITY_INTERRUPTING_IDLE (without sending DC_EVENT_CONNECTIVITY_CHANGED).
```

Comment:
Why is `DC_EVENT_CONNECTIVITY_CHANGED` not emitted?

Reply:
I explained it at https://github.com/deltachat/deltachat-core-rust/pull/2319/files#diff-44a04e55f5114b7bb7842524cf438cc27ae99a4242deeb147a2f46a48e9f4a62R154-R158.

It would be unnecessary from a technical point of view (would have no advantages, it's just DC_CONNECTIVITY_INTERRUPTING_IDLE so that it isn't DC_CONNECTIVITY_CONNECTED), but I'm currently wondering if we should just emit this event to reduce confusion for people reading this.